### DISCUSSION TITLE
[AIP-11] Create a round via program / registry

### DISCUSSION THREAD

**Discussion by thelostone-mc** [allo-protocol/aips#8] [AIPs in Review]
**Problem**

The current system allows for the creation of rounds within the program by individuals who may not be legitimate members. This lack of verification exposes the program to potential fraud and spoofing of rounds. It is essential to address this issue and ensure that only authorized users of the program can create rounds, especially considering that both the program and rounds are on the same chain.

What are we trying to solve?

- We aim to prevent unauthorized individuals from creating rounds within the program, thereby reducing the risk of fraudulent activities and spoofing.
- By enforcing the creation of rounds exclusively through the program, we can maintain the integrity and security of the program.
- Given that the program and rounds exist on the same chain, implementing this solution should be a relatively straightforward process

**Comment by 0xKurt** [comment]
> By enforcing the creation of rounds exclusively through the program, we can maintain the integrity and security of the program.

we def. should integrate it this way.
In addion i would like to add a list/mapping of program->round to be able to validate if a round really belongs to a progrm

**Comment by thelostone-mc** [comment]
The purpose of this revision is to consolidate the contract into a cohesive unit.

- The programFactory and programImplementation components are removed.
- Now, a project has the ability to create a round or apply to a round.

The advantages of this approach are as follows:
- Simplifies the implementation of a reputation system, as the registry holds projects.
- Eliminates the distinction between programs and projects. A program is now reimagined as a project that creates a round.
- Enables the reuse of existing roles in the project registry.
- Reduces costs and simplifies maintenance, as there are now two fewer contracts to manage.
- Ensures that only project owners with project metadata can create a round.

**Comment by thelostone-mc** [comment]
This has been implemented via https://github.com/allo-protocol/aips/pull/19

**Comment by nfrgosselin** [comment]
Re-opening this discussion for any ongoing thoughts on [AIP-11 ](https://github.com/allo-protocol/aips/pull/19)

**Reply by thelostone-mc** [reply]
Closing this out as we've got the approval from the team to go forward with this
